### PR TITLE
ci: add docker push in release ci

### LINKED ISSUE #198: The CustomResourceDefinition "peers.kilo.squat.ai" is invalid: spec.scope: Invalid value: "Namespaced": field is immutable
i try update kilo to leatest version but i fond there is one crd yaml neen to apply from readme 

```
kubectl apply -f https://raw.githubusercontent.com/squat/kilo/main/manifests/crds.yaml
kubectl apply -f https://raw.githubusercontent.com/squat/kilo/main/manifests/kilo-k3s.yaml
```

but i stuck in first step

```
Warning: resource customresourcedefinitions/peers.kilo.squat.ai is missing the kubectl.kubernetes.io/last-applied-configuration annotation which is required by kubectl apply. kubectl apply should only be used on resources created declaratively by either kubectl create --save-config or kubectl apply. The missing annotation will be patched automatically.
The CustomResourceDefinition "peers.kilo.squat.ai" is invalid: spec.scope: Invalid value: "Namespaced": field is immutable
```

my k3s version is `v1.21.1+k3s1`

could anyone can help me ?

**BTW**, i think kilo need docker image with release tag, i can do some hlep work if you need

### PR DESCRIPTION
**Pull request by Ehco1996**
close #198

### CODE REVIEW

**Comment by squat** [comment]
I wonder if to simplify this PR and have less duplicate code that we have to keep in sync, it would be easier if we could add `tag` as a trigger for the CI workflow, which also tags and pushes images. WDYT?

**Comment by Ehco1996** [comment]
> I wonder if to simplify this PR and have less duplicate code that we have to keep in sync, it would be easier if we could add `tag` as a trigger for the CI workflow, which also tags and pushes images. WDYT?

@squat  thanks for your review, push tag will also trigger this ci now

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -1,21 +1,52 @@
 on:
   release:
     types: [created]
+  push:
+    tags:
+      - "*"
 name: Handle Release
```
Comment (squat) [inline .github/workflows/release.yaml:7]: @Ehco1996 i meant actually *only* adding this trigger to the other workflow file and removing the content below. The reason is that the sections below are mostly duplicated code and now we always have to keep two CI files in sync. Whenever we change the image tagging code in ci.yaml we also need to update release.yaml, which is easy to forget and is likely going to cause issues. I would suggest we make the addition only in ci.yaml 👍

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -3,19 +3,19 @@ on:
...
-        go-version: 1.16.5
-    - name: Make Directory with kgctl Binaries to Be Released
-      run: make release
-    - name: Publish Release
-      uses: skx/github-action-publish-binaries@master
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      with:
-        args: 'bin/release/kgctl-*'
+      - uses: actions/checkout@v2
```
Comment (squat) [inline .github/workflows/release.yaml:9]: I think these white-space changes don't need to be in this PR. Please revert this file and then we can merge the PR :rocket:

Comment (Ehco1996) [reply .github/workflows/release.yaml]: fixed, many thanks for you

**Review by squat — APPROVED** [review]

**Comment by squat** [comment]
Thank you @Ehco1996!

**Comment by leonnicolas** [comment]
I think when we publish a new release over the github web frontend, ci.yaml will not run because no tag was pushed.

And if we push a new tag, release.yaml will not run (I don't know why, but I think this happend last time, even though github creates a new release for the pushed tags)

**Comment by squat** [comment]
Yes, this means that technically we won't publish new images on a release but rather we'll publish new images whenever a new tag is pushed to GitHub. I think that's fine tho. Do you see that maybe causing problems?

**Pull request merged**